projects/ngx-diagrams/src/lib/models/node.model.ts

```diff
@@ -1,31 +1,32 @@
 import { BehaviorSubject, Observable } from 'rxjs';
-import { PortModel } from './port.model';
-import { BaseModel } from './base.model';
-import { DiagramModel } from './diagram.model';
-import { Coords } from '../interfaces/coords.interface';
-import { DiagramEngine } from '../services/engine.service';
 import { map } from 'rxjs/operators';
+import { Coords } from '../interfaces/coords.interface';
 import { Dimensions } from '../interfaces/dimensions.interface';
-import { ID, mapToArray } from '../utils/tool-kit.util';
 import { SerializedNodeModel } from '../interfaces/serialization.interface';
+import { DiagramEngine } from '../services/engine.service';
+import { ID, mapToArray } from '../utils/tool-kit.util';
+import { HashMap } from '../utils/types';
+import { BaseModel } from './base.model';
+import { DiagramModel } from './diagram.model';
+import { PortModel } from './port.model';
 
 export class NodeModel<P extends PortModel = PortModel> extends BaseModel<DiagramModel> {
-	private readonly _diagramEngine$: BehaviorSubject<DiagramEngine> = new BehaviorSubject<DiagramEngine>(null);
-	private readonly _extras$: BehaviorSubject<{ [s: string]: any }> = new BehaviorSubject({});
-	private readonly _ports$: BehaviorSubject<{ [s: string]: P }> = new BehaviorSubject({});
-	private readonly _coords$: BehaviorSubject<Coords> = new BehaviorSubject<Coords>({ x: 0, y: 0 });
-	private readonly _dimensions$: BehaviorSubject<Dimensions> = new BehaviorSubject<Dimensions>({ width: 0, height: 0 });
-
-	private readonly diagramEngine$: Observable<{ [s: string]: any }> = this._diagramEngine$.pipe(this.entityPipe('diagramEngine'));
-	private readonly extras$: Observable<{ [s: string]: any }> = this._extras$.pipe(this.entityPipe('extras'));
-	private readonly ports$: Observable<{ [s: string]: P }> = this._ports$.pipe(this.entityPipe('ports'));
-	private readonly coords$: Observable<Coords> = this._coords$.pipe(this.entityPipe('coords'));
-	private readonly dimensions$: Observable<Dimensions> = this._dimensions$.pipe(this.entityPipe('dimensions'));
+	protected readonly _diagramEngine$ = new BehaviorSubject<DiagramEngine>(null);
+	protected readonly _extras$ = new BehaviorSubject<HashMap<any>>({});
+	protected readonly _ports$ = new BehaviorSubject<HashMap<P>>({});
+	protected readonly _coords$ = new BehaviorSubject<Coords>({ x: 0, y: 0 });
+	protected readonly _dimensions$ = new BehaviorSubject<Dimensions>({ width: 0, height: 0 });
+
+	protected readonly diagramEngine$ = this._diagramEngine$.asObservable().pipe(this.entityPipe('diagramEngine'));
```

Comment:
piping a behaviorSubject turns it into an observable automatically, using asObservable is redundant